source/System.h

```diff
@@ -172,6 +173,9 @@ class System {
 	// or links, figure out which stars are "neighbors" of this one, i.e.
 	// close enough to see or to reach via jump drive.
 	void UpdateNeighbors(const Set<System> &systems, double distance);
+	void ReadInt(const DataNode &node, const std::string &name, int &value, int index);
```

Comment:
This should not be a function of the system class.
Two options I propose:

-  function of the node class
- add it to an unnamed namespace in system.cpp

Reply:
I'll change some argument names and move it to Node.